From a80bc2ede2d11b40d1e6790413c4ebd7d828b7f0 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 28 Apr 2016 10:05:08 +0800
Subject: [PATCH 1029/5242] MLK-12724 ARM: dts: imx6ull-ddr3-arm2: resolve pin
 conflicts for pwm3 and lcd touch

commit  73182789cf20baa6082d2d847f7d7b53c8803a48 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since pwm3 and lcd touch are conflict for one pin, change
backlight to use pwm1 instead. But pwm1 also conflicts with
enet1, so enable backlight in a new dts file.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts |   46 ++----------------
 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts      |   50 +++++++++++++++++++-
 2 files changed, 53 insertions(+), 43 deletions(-)

diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts
index e4a8d0d..40e86a4 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts
@@ -11,55 +11,19 @@
 #include "imx6ull-14x14-ddr3-arm2.dts"
 
 / {
-	regulators {
-		reg_usb_otg1_vbus: regulator@2 {
-			status = "disabled";
-		};
-	};
-
 	backlight {
-		compatible = "pwm-backlight";
-		pwms = <&pwm3 0 5000000>;
-		brightness-levels = <0 4 8 16 32 64 128 255>;
-		default-brightness-level = <6>;
 		status = "okay";
 	};
 };
 
+&fec1 {
+	status = "disabled";
+};
+
 &lcdif {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lcdif_dat
-		     &pinctrl_lcdif_ctrl>;
-	display = <&display0>;
 	status = "okay";
-
-	display0: display {
-		bits-per-pixel = <16>;
-		bus-width = <24>;
-
-		display-timings {
-			native-mode = <&timing0>;
-			timing0: timing0 {
-				clock-frequency = <33500000>;
-				hactive = <800>;
-				vactive = <480>;
-				hback-porch = <89>;
-				hfront-porch = <164>;
-				vback-porch = <23>;
-				vfront-porch = <10>;
-				hsync-len = <10>;
-				vsync-len = <10>;
-				hsync-active = <0>;
-				vsync-active = <0>;
-				de-active = <1>;
-				pixelclk-active = <0>;
-			};
-		};
-	};
 };
 
-&pwm3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm3>;
+&pwm1 {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
index 94e494c..8486b14 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
@@ -23,6 +23,14 @@
 		reg = <0x80000000 0x40000000>;
 	};
 
+	backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm1 0 5000000>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <6>;
+		status = "disabled";
+	};
+
 	pxp_v4l2 {
 		compatible = "fsl,imx6ul-pxp-v4l2", "fsl,imx6sx-pxp-v4l2", "fsl,imx6sl-pxp-v4l2";
 		status = "okay";
@@ -644,9 +652,9 @@
 			>;
 		};
 
-		pinctrl_pwm3: pmw3grp {
+		pinctrl_pwm1: pmw1grp {
 			fsl,pins = <
-				MX6UL_PAD_GPIO1_IO04__PWM3_OUT	0x110b0
+				MX6UL_PAD_ENET1_RX_DATA0__PWM1_OUT   0x110b0
 			>;
 		};
 
@@ -852,6 +860,44 @@
 	};
 };
 
+&lcdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lcdif_dat
+		     &pinctrl_lcdif_ctrl>;
+	display = <&display0>;
+	status = "disabled";
+
+	display0: display {
+		bits-per-pixel = <16>;
+		bus-width = <24>;
+
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: timing0 {
+				clock-frequency = <33500000>;
+				hactive = <800>;
+				vactive = <480>;
+				hback-porch = <89>;
+				hfront-porch = <164>;
+				vback-porch = <23>;
+				vfront-porch = <10>;
+				hsync-len = <10>;
+				vsync-len = <10>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <1>;
+				pixelclk-active = <0>;
+			};
+		};
+	};
+};
+
+&pwm1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "disabled";
+};
+
 &pxp {
 	status = "okay";
 };
-- 
1.7.9.5

